libpod/container_internal_linux.go

```diff
@@ -174,24 +174,35 @@ func (c *Container) prepare() error {
 		return err
 	}
 
-	// Ensure container entrypoint is created (if required)
+	workdir := c.WorkingDir()
+	resolvedWorkdir, err := securejoin.SecureJoin(c.state.Mountpoint, c.WorkingDir())
+	if err != nil {
+		return errors.Wrapf(err, "error determining workdir path on container %s", c.ID())
+	}
 	if c.config.CreateWorkingDir {
-		workdir, err := securejoin.SecureJoin(c.state.Mountpoint, c.WorkingDir())
-		if err != nil {
-			return errors.Wrapf(err, "error creating path to container %s working dir", c.ID())
-		}
+		// Ensure container entrypoint is created (if required).
 		rootUID := c.RootUID()
 		rootGID := c.RootGID()
 
-		if err := os.MkdirAll(workdir, 0755); err != nil {
+		if err := os.MkdirAll(resolvedWorkdir, 0755); err != nil {
 			if os.IsExist(err) {
 				return nil
 			}
-			return errors.Wrapf(err, "error creating container %s working dir", c.ID())
+			return errors.Wrapf(err, "error creating container %s workdir", c.ID())
 		}
 
-		if err := os.Chown(workdir, rootUID, rootGID); err != nil {
-			return errors.Wrapf(err, "error chowning container %s working directory to container root", c.ID())
+		if err := os.Chown(resolvedWorkdir, rootUID, rootGID); err != nil {
+			return errors.Wrapf(err, "error chowning container %s workdir to container root", c.ID())
+		}
+	} else {
+		// Otherwise, make sure the workdir is present.
+		if _, err := os.Stat(resolvedWorkdir); err != nil {
+			if os.IsNotExist(err) {
+				return errors.Errorf("workdir %q does not exist on container %s", workdir, c.ID())
```

Comment:
```suggestion
				return errors.Errorf("workdir %q does not exist on container %s", resolvedWorkdir, c.ID())
```

Reply:
I don't think the resolved one is helpful.  That'll be the fully resolved path on the host which I think can easily confuse users.  I found nicer to return the erroneous input so users can match the error to their input.

Reply:
Ignore this and the next if you wish.  I saw the workdir initialization removed from the old code, but missed it being added at the very top in this new change.  I'll trust you on which is the better to present here to the user and down below.